zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java

```diff
@@ -66,9 +66,9 @@ public void registerWith(final StubbedBrokerClient gateway) {
   @Override
   public BrokerResponse<DecisionEvaluationRecord> handle(
       final BrokerEvaluateDecisionRequest request) throws Exception {
-    // In practise the decision instance key is generated newly by the broker
-    final long decisionInstanceKey = DECISION_RECORD.getDecisionKey() + 1;
-    return new BrokerResponse<>(DECISION_RECORD, request.getPartitionId(), decisionInstanceKey);
+    final DecisionEvaluationRecord evaluationRecord = createDecisionRecord();
+    return new BrokerResponse<>(
+        evaluationRecord, request.getPartitionId(), evaluationRecord.getDecisionKey());
```

Comment:
💭 Previously, this effectively was `evaluationRecord.getDecisionKey() + 1`. I am still trying to understand why and if this needs to be adjusted here.